app/controllers/assessments_controller.rb

```diff
@@ -587,6 +604,30 @@ def viewFeedback
     }
   end
 
+  action_auth_level :getPartialFeedback, :student
+
+  def getPartialFeedback
+    job_id = params["job_id"].to_i
+    resp = {}
+    # User requested to view feedback on a score
+    if job_id.nil?
+      flash[:error] = "Invalid job id"
+      redirect_to(action: "index") && return
+    end
+
+    begin
+      resp['partial_feedback'] = tango_get_partial_feedback(job_id)
+    rescue AutogradeError
+      # if unable to get partial feedback, check if job is on the queue
+      @job_status = get_job_status(job_id)
+      resp["is_assigned"] = @job_status["is_assigned"]
+      resp["queue_position"] = @job_status["queue_position"]
+      resp["queue_length"] = @job_status["queue_length"]
+    end
+
+    render json: resp.to_json
```

Comment:
So relevant to the other situation with regards to the refresh loop, you should just throw an error instead of return partial feedback

Reply:
flash[:error] & redirect were not working... added an error message from frontend to handle this instead